dom.go

```diff
@@ -418,20 +454,26 @@ func renderComponent(comp Component, prevRender *HTML) (h *HTML, skip bool) {
 		nextRender = Tag("noscript")
 	}
 
-	// Restore the actual rendered HTML.
+	// Reconcile the actual rendered HTML.
 	nextRender.reconcile(prevRender)
 
 	// Update the context to consider this render.
-	comp.Context().prevRender = nextRender
-	comp.Context().prevComponent = comp
-	comp.Context().prevRenderComponent = doCopy(comp)
+	//
+	// TODO: In the event that this really is the previous component instance,
```

Comment:
I'm like 80% certain this is the cause of the issue mentioned in this TODO comment of my test file: https://github.com/gopherjs/vecty/commit/7fc6935b2ca7722641a4e2fa90dc30d06623697f#diff-59654adfef9097fef9e9fede9315019aR45

The overarching problem here is that, now that we have persistent component references, it's actually not always obvious where the proper `Context()` really is. Someone will need to sit down and rethink how to fix this issue here + hopefully make the `Context` storage location a bit easier to think about..